heat/utils/data/mnist.py

```diff
@@ -0,0 +1,126 @@
+import torch
+
+from PIL import Image
+from heat.core import factories
+from torchvision import datasets
+from typing import Callable, Union
+
+__all__ = ["MNISTDataset"]
+
+
+class MNISTDataset(datasets.MNIST):
+    """
+    Dataset wrapper for :class:`torchvision.datasets.MNIST`. This implements all of the required functions mentioned in
+    :class:`heat.utils.data.Dataset`. The ``__getitem__`` and ``__len__`` functions are inherited from
+    :class:`torchvision.datasets.MNIST`.
+
+    Parameters
+    ----------
+    root : str
```

Comment:
this is how torch does it. essentially, the MNIST data is downloaded to a directory if it doesnt exist there already. then the data is loaded from the folder. since the data is small, it can be loaded fully into memory before being split. but this is a special case for this specific dataset. For larger datasets, a Dataset class needs to be defined.

Reply:
Then we should stick to it